index.js

```diff
@@ -99,6 +99,16 @@ class HtmlWebpackPlugin {
       this.options.filename = path.relative(compiler.options.output.path, filename);
     }
 
+    // `contenthash` is introduced in webpack v4.3
+    // which conflicts with the plugin's existing `contenthash` method, hence
+    // hence is renmaed to `templatehash` to avoid such conflicts
+    const REGEXP_CONTENTHASH = /\[(?:(\w+):)?contenthash(?::([a-z]+\d*))?(?::(\d+))?\]/ig;
```

Comment:
I think this could be simplified to:

```
      this.options.filename = this.options.filename.replace(REGEXP_CONTENTHASH, (match) => {
        return match.replace('contenthash', 'templatehash');
      });
```

No need for the verbose variable declaration and the if check.

Reply:
Yeah, I thought of moving them both to the very top. hence assigned to a variable. Which I thought may be reusable in future.